etc/di.xml

```diff
@@ -110,50 +101,56 @@
 
     <virtualType name="OmiseOffsiteAlipayInitializeCommand" type="Magento\Payment\Gateway\Command\GatewayCommand">
         <arguments>
-            <argument name="requestBuilder" xsi:type="object">OmiseOffsiteAlipayRequest</argument>
+            <argument name="requestBuilder" xsi:type="object">OmiseOffsiteRequest</argument>
             <argument name="transferFactory" xsi:type="object">Omise\Payment\Gateway\Http\TransferFactory</argument>
             <argument name="client" xsi:type="object">OmiseCharge</argument>
             <argument name="handler" xsi:type="object">OmiseOffsiteResponseHandler</argument>
             <argument name="validator" xsi:type="object">Omise\Payment\Gateway\Validator\Offsite\AlipayInitializeCommandResponseValidator</argument>
         </arguments>
     </virtualType>
+    <!-- /Command Pool -->
 
-    <virtualType name="OmiseOffsiteAlipayRequest" type="Magento\Payment\Gateway\Request\BuilderComposite">
+    <!-- Offsite Request -->
+    <virtualType name="OmiseOffsiteRequest" type="Magento\Payment\Gateway\Request\BuilderComposite">
         <arguments>
             <argument name="builders" xsi:type="array">
                 <item name="payment" xsi:type="string">Omise\Payment\Gateway\Request\PaymentDataBuilder</item>
                 <item name="offsite" xsi:type="string">Omise\Payment\Gateway\Request\PaymentOffsiteBuilder</item>
             </argument>
         </arguments>
     </virtualType>
+    <!-- /Offsite Request -->
 
-    <virtualType name="OmiseResponseHandler" type="Magento\Payment\Gateway\Response\HandlerChain">
+    <!-- Offsite Response Handlers -->
+     <virtualType name="OmiseOffsiteResponseHandler" type="Magento\Payment\Gateway\Response\HandlerChain">
         <arguments>
             <argument name="handlers" xsi:type="array">
                 <item name="paymentDetails" xsi:type="string">Omise\Payment\Gateway\Response\PaymentDetailsHandler</item>
+                <item name="pendingPayment" xsi:type="string">Omise\Payment\Gateway\Response\PendingPaymentHandler</item>
+                <item name="pendingInvoice" xsi:type="string">Omise\Payment\Gateway\Response\PendingInvoiceHandler</item>
             </argument>
         </arguments>
     </virtualType>
 
-    <virtualType name="OmiseOffsiteResponseHandler" type="Magento\Payment\Gateway\Response\HandlerChain">
```

Comment:
this part of the code was moved bit below, and grouped with Offsite Hanlders, that is why it shows falsely that class was renamed.